src/test/java/plugins/LdapPluginTest.java

```diff
@@ -205,7 +249,7 @@ public void login_use_fallback_server() {
         int freePort = this.findAvailablePort();
         LdapDetails ldapDetails = new LdapDetails("", 0, ldapContainer.getManagerDn(), ldapContainer.getManagerPassword(), ldapContainer.getRootDn());
         // Fallback-Config: primary server is not running, alternative server is running docker fixture
-        ldapDetails.setHostWithPort("localhost:" + freePort + " localhost:" + ldapContainer.getPort());
+        ldapDetails.setHostWithPort(ldapContainer.getHost()+":" + freePort + " " + ldapContainer.getHost() + ":" + ldapContainer.getPort());
```

Comment:
> Unless I am misunderstanding something

Yes.  You are missing the fact that this does [`findAvailablePort()`](https://github.com/jenkinsci/acceptance-test-harness/blob/caddb01a1b21bf19f12de6cd8cf687b63bafcec2/src/main/java/org/jenkinsci/test/acceptance/junit/AbstractJUnitTest.java#L45-L54) which will find a port that is not in use on the local system so there is nothing running on that.  It also binds to it and does not set SO_REUSE_ADDR so it should not be re-cycled for a period of time (which should be longer than the time it takes to run the tests).

> I think it works just fine in both cases, using ldapContainer.getHost() or localhost. What matters is the port.

Not true - networking plays a role here

The `ldapContainer.getHost()` will return the host of the docker process that you are talking to (this may not be localhost).
You can not guarantee that there is not something listening on that specific port on a remote machine.